racket-repl.el

```diff
@@ -526,79 +527,98 @@ WHAT-TO-RUN may be nil, meaning just a `racket/base` namespace."
-             ;; Buffer might already be in `racket-repl-mode' -- e.g.
-             ;; `racket-repl-exit' was used and now we're
-             ;; "restarting". In that case avoid re-initialization
-             ;; that is at best unnecessary or at worst undesirable
-             ;; (e.g. `comint-input-ring' would lose input history).
-             (unless (eq major-mode 'racket-repl-mode)
-               (when noninteractive
-                 (princ "{racket--repl-start}: (racket-repl-mode)\n"))
-               (racket-repl-mode)))
-         (file-error
-          (let ((kill-buffer-query-functions nil)
-                (kill-buffer-hook nil))
-            (kill-buffer)) ;don't leave partially initialized REPL buffer
-          (message "Could not connect to REPL server at 127.0.0.1:%s" repl-tcp-port-number)))))))
+                        (princ (format "{racket--repl-start}: %s\n" id)))
+                      (setq racket--repl-session-id id)
+                      (run-with-timer 0.001 nil callback)
+                      (remove-hook 'comint-preoutput-filter-functions hook t)
+                      (prog1
+                          (with-current-buffer read-buf
+                            (buffer-substring (if (eq (char-after) ?\n)
+                                                  (1+ (point))
+                                                (point))
+                                              (point-max)))
+                        (kill-buffer read-buf)))
+                     (_ ""))))
+      (add-hook 'comint-preoutput-filter-functions hook nil t))
+
+    (condition-case ()
+        (progn
+          (make-comint-in-buffer racket-repl-buffer-name
+                                 (current-buffer)
+                                 (cons repl-host-name repl-tcp-port-number))
+          (process-send-string (get-buffer-process (current-buffer))
+                               (format "\"%s\"\n" racket--cmd-auth))
+          (when noninteractive
+            (princ "{racket--repl-start}: did process-send-string of auth\n"))
+          (set-process-coding-system (get-buffer-process (current-buffer))
+                                     'utf-8 'utf-8) ;for e.g. λ
+          ;; Buffer might already be in `racket-repl-mode' -- e.g.
+          ;; `racket-repl-exit' was used and now we're
+          ;; "restarting". In that case avoid re-initialization
+          ;; that is at best unnecessary or at worst undesirable
+          ;; (e.g. `comint-input-ring' would lose input history).
+          (unless (eq major-mode 'racket-repl-mode)
+            (when noninteractive
+              (princ "{racket--repl-start}: (racket-repl-mode)\n"))
+            (racket-repl-mode)))
+      (file-error
+       (let ((kill-buffer-query-functions nil)
+             (kill-buffer-hook nil))
+         (kill-buffer)) ;don't leave partially initialized REPL buffer
+       (message "Could not connect to REPL server: %s"
```

Comment:
I think this is the format string the bytecode compiler is complaining about? https://github.com/greghendershott/racket-mode/actions/runs/1117388372